src/services/commandExecutor.ts

```diff
@@ -25,15 +23,11 @@ export interface IGithubCommandExecutor {
 
 export class ShellCommandExecutor implements ICommandExecutor {
     async execute(commands: string[]): Promise<CommandExecutorResponse> {
```

Comment:
It doesn't strictly need to be - it might make more semantic sense to remove the `async`, but it'll be the same functional execution. 

To be honest, I'm reluctant to change the return signature here - I'd rather keep as much behavior as possible the same and make this change in isolation.